bitnami/kafka/templates/_helpers.tpl

```diff
@@ -346,19 +353,457 @@ Return the Kafka configuration configmap
+ssl.truststore.type=JKS
+ssl.keystore.location=/opt/bitnami/kafka/config/certs/kafka.keystore.jks
+ssl.truststore.location=/opt/bitnami/kafka/config/certs/kafka.truststore.jks
+#ssl.keystore.password=
+#ssl.truststore.password=
+#ssl.key.password=
+ssl.client.auth={{ .Values.tls.sslClientAuth }}
+ssl.endpoint.identification.algorithm={{ .Values.tls.endpointIdentificationAlgorithm }}
+{{- end }}
+{{- if .Values.acl.authorizerClassName }}
+authorizer.class.name={{ .Values.acl.authorizerClassName }}
+{{- end }}
+{{- if .Values.acl.superUsers }}
+super.users={{ .Values.acl.superUsers }}
+{{- end }}
+{{- if not (empty .Values.acl.allowEveryoneIfNoAclFound) }}
+allow.everyone.if.no.acl.found={{ .Values.acl.allowEveryoneIfNoAclFound }}
+{{- end }}
+{{- if (include "kafka.saslEnabled" .) }}
+# Listeners SASL JAAS configuration
+{{- $listeners := list .Values.listeners.client .Values.listeners.interbroker }}
+{{- range $i := .Values.listeners.extraListeners }}
+{{- $listeners = append $listeners $i }}
+{{- end }}
+{{- range $listener := $listeners }}
+{{- if and $listener.sslClientAuth (regexFind "SSL" (upper $listener.protocol)) }}
+listener.name.{{lower $listener.name}}.ssl.client.auth={{ $listener.sslClientAuth }}
+{{- end }}
+{{- if regexFind "SASL" (upper $listener.protocol) }}
+{{- range $mechanism := ( splitList "," $.Values.sasl.enabledMechanisms )}}
+  {{- $securityModule := ternary "org.apache.kafka.common.security.plain.PlainLoginModule required" "org.apache.kafka.common.security.scram.ScramLoginModule required" (eq "PLAIN" (upper $mechanism)) }}
+  {{- $saslJaasConfig := list $securityModule }}
+  {{- if eq $listener.name $.Values.listeners.interbroker.name }}
+  {{- $saslJaasConfig = append $saslJaasConfig (printf "username=\"%s\"" $.Values.sasl.interbroker.user) }}
+  {{- $saslJaasConfig = append $saslJaasConfig (print "password=\"interbroker-password-placeholder\"") }}
+  {{- end }}
+  {{- if eq (upper $mechanism) "PLAIN" }}
+  {{- if eq $listener.name $.Values.listeners.interbroker.name }}
+  {{- $saslJaasConfig = append $saslJaasConfig (printf "user_%s=\"interbroker-password-placeholder\"" $.Values.sasl.interbroker.user) }}
+  {{- end }}
+  {{- range $i, $user := $.Values.sasl.client.users }}
+  {{- $saslJaasConfig = append $saslJaasConfig (printf "user_%s=\"password-placeholder-%d\"" $user (int $i)) }}
+  {{- end }}
+  {{- end }}
+listener.name.{{lower $listener.name}}.{{lower $mechanism}}.sasl.jaas.config={{ join " " $saslJaasConfig }};
+{{- end }}
+{{- end }}
+{{- end }}
+# End of SASL JAAS configuration
+{{- end }}
+{{- end -}}
+
+{{/*
+Zookeeper connection section of the server.properties
+*/}}
+{{- define "kafka.zookeeperConfig" -}}
+zookeeper.connect={{ include "kafka.zookeeperConnect" . }}
+#broker.id=
+{{- if .Values.sasl.zookeeper.user }}
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
+    username="{{ .Values.sasl.zookeeper.user }}" \
+    password="zookeeper-password-placeholder";
```

Comment:
FWIW, In other solutions we use `render-template`, mapping it to an env var and thus not having to perform logic like `sed` in init containers. Not sure if kafka has `render-template`

Reply:
In this case, init-container does not use `bitnami/kafka` container but `bitnami/bitnami-shell` container which does not include `render-template`.